### PR TITLE
[ISSUE #4164] Remove unnecessary properties from message when messageTimeup

### PR DESCRIPTION
**Pull request by BurningCN**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

issue #4164

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: store/src/main/java/org/apache/rocketmq/store/schedule/ScheduleMessageService.java [hunk 1]
```diff
@@ -327,14 +326,18 @@ private MessageExtBrokerInner messageTimeup(MessageExt msgExt) {
...
 
         String queueIdStr = msgInner.getProperty(MessageConst.PROPERTY_REAL_QUEUE_ID);
         int queueId = Integer.parseInt(queueIdStr);
         msgInner.setQueueId(queueId);
 
+        MessageAccessor.clearProperty(msgInner, MessageConst.PROPERTY_DELAY_TIME_LEVEL);
+        MessageAccessor.clearProperty(msgInner, MessageConst.PROPERTY_REAL_TOPIC);
+        MessageAccessor.clearProperty(msgInner, MessageConst.PROPERTY_REAL_QUEUE_ID);
+        MessageAccessor.setProperties(msgInner, msgExt.getProperties());
+
```
Comment (hzh0425) [inline store/src/main/java/org/apache/rocketmq/store/schedule/ScheduleMessageService.java:340]: It looks like you should clear the properties of msgExt and then set the properties in msgExt to msgInner

Comment (BurningCN) [reply store/src/main/java/org/apache/rocketmq/store/schedule/ScheduleMessageService.java:340]: ok, thanks

**Comment by hzh0425** [comment]
The ci test was failed, you should check in your local first

**Comment by dongeforever** [comment]
@BurningCN The additional properties are useful for debugging.
So it is better to keep it.

File: store/src/main/java/org/apache/rocketmq/store/schedule/ScheduleMessageService.java [hunk 2]
```diff
@@ -327,14 +325,19 @@ private MessageExtBrokerInner messageTimeup(MessageExt msgExt) {
...
 
-        msgInner.setTopic(msgInner.getProperty(MessageConst.PROPERTY_REAL_TOPIC));
+        msgInner.setTopic(msgExt.getProperty(MessageConst.PROPERTY_REAL_TOPIC));
 
-        String queueIdStr = msgInner.getProperty(MessageConst.PROPERTY_REAL_QUEUE_ID);
+        String queueIdStr = msgExt.getProperty(MessageConst.PROPERTY_REAL_QUEUE_ID);
         int queueId = Integer.parseInt(queueIdStr);
         msgInner.setQueueId(queueId);
 
+        MessageAccessor.clearProperty(msgExt, MessageConst.PROPERTY_DELAY_TIME_LEVEL);
```
Comment (lwclover) [inline store/src/main/java/org/apache/rocketmq/store/schedule/ScheduleMessageService.java:335]: As a good programming practice, I don't think the contents of the method argument msgExt should change.
Here's my advice:
MessageAccessor.setProperties(msgInner, msgExt.getProperties());
MessageAccessor.clearProperty(msgInner, MessageConst.PROPERTY_REAL_TOPIC);
.......
msgInner.setPropertiesString(MessageDecoder.messageProperties2String(msgInner.getProperties()));

**Review by lwclover — CHANGES_REQUESTED** [review]

**Comment by BurningCN** [comment]
> 


> @BurningCN The additional properties are useful for debugging. So it is better to keep it.

OK

File: store/src/main/java/org/apache/rocketmq/store/schedule/ScheduleMessageService.java (hunk 2 above)
Comment (BurningCN) [reply store/src/main/java/org/apache/rocketmq/store/schedule/ScheduleMessageService.java:335]: > As a good programming practice, I don't think the contents of the method argument msgExt should change. Here's my advice: MessageAccessor.setProperties(msgInner, msgExt.getProperties()); MessageAccessor.clearProperty(msgInner, MessageConst.PROPERTY_REAL_TOPIC); ....... msgInner.setPropertiesString(MessageDecoder.messageProperties2String(msgInner.getProperties()));

thanks

**Pull request closed without merging**